### PR TITLE
Make sure Content-Length is adjusted

### PR DESCRIPTION
**Pull request by sj26**
When rewriting input to sanitize the encoding sometimes the bytesize might be slightly changed, causing middleware and apps which respect the request content length to misread the input. This adjusts the content length when present.

### CODE REVIEW

File: lib/rack/utf8_sanitizer.rb [hunk 1]
```diff
@@ -62,7 +62,13 @@ def sanitize_rack_input(env)
       content_type &&= content_type.downcase
       return unless SANITIZABLE_CONTENT_TYPES.any? {|type| content_type == type }
       uri_encoded = URI_ENCODED_CONTENT_TYPES.any? {|type| content_type == type}
-      env['rack.input'] &&= sanitize_io(env['rack.input'], uri_encoded)
+
+      if env["rack.input"]
+        sanitized_input = sanitize_io(env['rack.input'], uri_encoded)
+
+        env['rack.input'] = sanitized_input
+        env['CONTENT_LENGTH'] &&= sanitized_input.size.to_s
```
Comment (bf4) [inline lib/rack/utf8_sanitizer.rb]: Tricky. I started to comment here that this should be `bytesize` but I see below in `#size` and in the tests that it already is. :100:

**Comment by aripollak** [comment]
Any chance of a new release with this fix?

**Comment by bf4** [comment]
@whitequark ?

**Comment by whitequark** [comment]
Sure. @bf4, can you do it?

**Comment by bf4** [comment]
Sure, I forgot I had push. Updating the CHANGELOG

**Comment by bf4** [comment]
I released it in 1.3.1.  I also created a PR to add Ruby 2.2 support (and fixed a test bug, that has no impact on the release https://github.com/whitequark/rack-utf8_sanitizer/pull/28 )

**Pull request merged**